### ISSUE TITLE
Misleading script check error and Docker missing container exit

### ISSUE THREAD

**Issue by schmichael** [hashicorp/nomad#6110]
### Nomad version

0.9.4

### Issue

_Submitting a report based on a chat with a friend._

There may be 2 issues:

1. Docker task may have exited without being noticed by Nomad. The "driver doesn't support..." error at 15:02:35 seems to indicate the driver handle is nil, but there are no task events related to the Docker container exiting.
2. The "driver doesn't support script checks" error may be erroneously emitted when a task is updated before it begins or during a restart (whenever the driver handle can be nil).

The user restarted the task at 20:03:28 after being alerted by a user that the service was down.

```
Time                  Type              Description
2019-08-08T20:03:34Z  Started           Task started by client
2019-08-08T20:03:34Z  Restarting        Task restarting in 0s
2019-08-08T20:03:34Z  Terminated        Exit Code: 137, Exit Message: "Docker container exited with non-zero exit code: 137"
2019-08-08T20:03:28Z  Restart Signaled  User requested restart
2019-08-08T15:02:35Z  Task hook failed  consul_services: driver doesn't support script checks
2019-07-31T05:56:37Z  Started           Task started by client
2019-07-31T05:55:54Z  Driver            Downloading image
2019-07-31T05:55:54Z  Task hook failed  consul_services: driver doesn't support script checks
2019-07-31T05:55:54Z  Task Setup        Building Task Directory
2019-07-31T05:55:39Z  Received          Task received by client
```

### Reproduction steps

1. Run a Docker job with a script check
2. Regularly update it
3. Look for `consul_services: driver doesn't support script checks` errors

### Nomad Client logs (if appropriate)

The only available logs seem to show an agent restart (nomad pid changes) followed by an inability to reconnect to the task.

In this circumstance the script check error is expected because the driver handle will be nil, but misleading because the *driver* does support script checks.

```
Aug  8 15:02:25 nomad-client-a1edce1e.internal nomad[4376]:     2019-08-08T15:02:25.633Z [WARN ] client.alloc_runner.task_runner.task_hook: failed to reattach to logmon process: alloc_id=c276ebd8-4afe-a12a-193d-35fdfa42b5fc task=taskname error="Reattachment process not found"
Aug  8 15:02:33 nomad-client-a1edce1e.internal nomad[7632]:     2019-08-08T15:02:33.243Z [INFO ] client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=c276ebd8-4afe-a12a-193d-35fdfa42b5fc task=taskname @module=logmon path=/var/lib/nomad/data/alloc/c276ebd8-4afe-a12a-193d-35fdfa42b5fc/alloc/logs/.taskname.stdout.fifo timestamp=2019-08-08T15:02:33.239Z
Aug  8 15:02:33 nomad-client-a1edce1e.internal nomad[7632]:     2019-08-08T15:02:33.243Z [INFO ] client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=c276ebd8-4afe-a12a-193d-35fdfa42b5fc task=taskname path=/var/lib/nomad/data/alloc/c276ebd8-4afe-a12a-193d-35fdfa42b5fc/alloc/logs/.taskname.stderr.fifo @module=logmon timestamp=2019-08-08T15:02:33.239Z
Aug  8 15:02:33 nomad-client-a1edce1e.internal nomad[7632]:     2019-08-08T15:02:33.149Z [WARN ] client.alloc_runner.task_runner.task_hook: failed to reattach to logmon process: alloc_id=c276ebd8-4afe-a12a-193d-35fdfa42b5fc task=taskname error="Reattachment process not found"
Aug  8 15:02:35 nomad-client-a1edce1e.internal nomad[7632]:     2019/08/08 15:02:35.626389 [INFO] (runner) rendered "(dynamic)" => "/var/lib/nomad/data/alloc/c276ebd8-4afe-a12a-193d-35fdfa42b5fc/taskname/secrets/.env"
Aug  8 15:02:37 nomad-client-a1edce1e.internal nomad[7632]:     2019-08-08T15:02:37.859Z [ERROR] client.alloc_runner.task_runner: update hook failed: alloc_id=c276ebd8-4afe-a12a-193d-35fdfa42b5fc task=taskname name=consul_services error="driver doesn't support script checks"
```

**Comment by tgross** [comment]
Thanks for this report! I spent much of this afternoon trying to reproduce this without success on Nomad 0.9.4 and Nomad 0.9.5. I think I have some insights as to the conditions though, so I learned a lot about logmon at least! The `failed to reattach to logmon process` error combined with the PID switch can be replicated as follows:

- start a job
- shut down the client agent: the Docker container is left running and `logmon`/`docker_logger` will be reparented to PID1.
- kill `logmon`
- restart the agent

That being said, in Nomad 0.10.0-beta (#6197) we've moved the script check lifecycle into its own task runner hook. Tasks that don't have script checks still have script check _hooks_ so they can be added by an update. Because of this, the line that emits this message in 0.9.4 and 0.9.5 is now at [`script_check_hook.go#L99`](https://github.com/hashicorp/nomad/blob/31f91cf4c8097cf65b08fd219ca9db692df66bbd/client/allocrunner/taskrunner/script_check_hook.go#L99) and is no longer an error. I think this issue is safe to close.
